app/PaymentDrivers/Stripe/SOFORT.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/**
+ * Invoice Ninja (https://invoiceninja.com)
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2020. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://opensource.org/licenses/AAL
+ */
+
+namespace App\PaymentDrivers\Stripe;
+
+use App\Events\Payment\PaymentWasCreated;
+use App\Jobs\Util\SystemLogger;
+use App\Models\GatewayType;
+use App\Models\PaymentType;
+use App\Models\SystemLog;
+use App\PaymentDrivers\StripePaymentDriver;
+
+class SOFORT
+{
+    public $stripe;
+
+    public function __construct(StripePaymentDriver $stripe)
+    {
+        $this->stripe = $stripe;
+    }
+
+    public function paymentView(array $data)
+    {
+        $data['gateway'] = $this->stripe;
+        $data['return_url'] = $this->buildReturnUrl($data);
+
+        return render('gateways.stripe.sofort.pay', $data);
+    }
+
+    private function buildReturnUrl($data): string
+    {
+        return route('client.payments.response', [
+            'company_gateway_id' => $this->stripe->company_gateway->id,
+            'gateway_type_id' => GatewayType::SOFORT,
+            'hashed_ids' => implode(",", $data['hashed_ids']),
+            'amount' => $data['amount'],
+            'fee' => $data['fee'],
+        ]);
+    }
+
+    public function paymentResponse($request)
+    {
+        $state = array_merge($request->all(), []);
+        $amount = $state['amount'] + $state['fee'];
+        $state['amount'] = $this->stripe->convertToStripeAmount($amount, $this->stripe->client->currency()->precision);
+
+        if ($request->redirect_status == 'succeeded') {
+            return $this->processSuccessfulPayment($state);
+        }
+
+        return $this->processUnsuccessfulPayment($state);
+    }
+
+    public function processSuccessfulPayment($state)
+    {
+        $state['charge_id'] = $state['source'];
+
+        $this->stripe->init();
+
+        $state['payment_type'] = PaymentType::SOFORT;
+
+        $data = [
+            'payment_method' => $state['charge_id'],
+            'payment_type' => $state['payment_type'],
+            'amount' => $state['amount'],
+        ];
+
+        $payment = $this->stripe->createPayment($data);
+
+        $this->stripe->attachInvoices($payment, $state['hashed_ids']);
+
+        $payment->service()->updateInvoicePayment();
```

Comment:
This is for discussion, in V2 we allow a payment to be made without an invoice present.... so there may not be any invoices present which will cause this method to fail.